_posts/2023-10-03-securing-wildfly-apps-oidc-openshift.adoc

```diff
@@ -0,0 +1,360 @@
+to check its status. Once it is running, you can click on *Open URL* and then access Keycloak's *Administration Console*.
+
+== Configure Keycloak
+
+. Log into the Keycloak Admin Console using the username and password you specified earlier.
+
+. Create a new realm called *myrealm*. For more information, see the Keycloak documentation on how to https://www.keycloak.org/getting-started/getting-started-openshift#_create_a_realm[create a realm].
+
+. Add a role called *user*. This role will be required to access our simple web application. For more information, see the Keycloak documentation on how to https://www.keycloak.org/docs/latest/server_admin/index.html#assigning-permissions-using-roles-and-groups[create a role].
+
+. Add a new user named *alice*. Set an *email* address for this new user, we'll use *alice@example.org*. For more information, see the Keycloak documentation on how to https://www.keycloak.org/getting-started/getting-started-openshift#_create_a_user[create a user].
+
+. Once the new user has been created, set a password for this new user from the *Credentials* tab.
+
+. From the *Role Mapping* tab, assign *alice* the *user* role. For more information, see the Keycloak documentation on how to https://www.keycloak.org/docs/latest/server_admin/index.html#proc-assigning-role-mappings_server_administration_guide[assign a role] to a user.
+
+. Create a new client as follows:
+* *General Settings*:
+** *Client type* (or *Client Protocol*, depending on your Keycloak version): *OpenID Connect*
+** *Client ID*: *myclient*
+* *Capability config*:
+** *Authentication flow*: *Standard flow*, *Direct access grants*
+* *Login settings*: Leave the fields blank for now.
+
++
+For more information, see the Keycloak documentation on how to https://www.keycloak.org/docs/latest/server_admin/index.html#_oidc_clients[Managing OpenID Connect clients].
+
+. Click *Save* to save the client. 
+
+== Add Helm Configuration
+
+. Obtain the URL for Keycloak.
++
+[source]
+----
+KEYCLOAK_URL=https://$(oc get route keycloak --template='{{ .spec.host }}') &&
+echo "" &&
+echo "Keycloak URL:   $KEYCLOAK_URL" &&
+echo ""
+----
+
+. Switch to the `charts` directory in the `simple-webapp-oidc` example.
++
+[source]
+----
+cd /PATH/TO/ELYTRON/EXAMPLES/simple-webapp-oidc/charts
+----
++
+Notice there's a `helm.yaml` file in this directory with the following content:
++
+[source]
+----
+build:
+  uri: https://github.com/wildfly-security-incubator/elytron-examples.git
+  contextDir: simple-webapp-oidc
+deploy:
+  replicas: 1
+  env:
+    - name: OIDC_PROVIDER_URL
+      value: <KEYCLOAK_URL>    <1>
+----
+<1> Replace <KEYCLOAK_URL> with the Keycloak URL obtained in the previous command.
+
+== Deploy the Example Application to WildFly on OpenShift
+
+We can deploy our example application to WildFly on OpenShift using the WildFly Helm Chart:
+
+[source]
+----
+helm install oidc-app -f /PATH/TO/ELYTRON/EXAMPLES/simple-webapp-saml/charts/helm.yaml wildfly/wildfly
```

Comment:
Would it be useful to add `helm repo add wildfly https://docs.wildfly.org/wildfly-charts/` before this? 
If this is their first time using the WildFly helm chart repo, they would not have it added and would get an error.

Reply:
That's a good question. We do mention WildFly Helm Chart in the pre-requisites and link to this page where adding the repo is mentioned:

https://docs.wildfly.org/30/Getting_Started_on_OpenShift.html#helm-charts

It's also important to update the WildFly Helm Chart if you've already added it. We should probably create an issue for mentioning that on the helm-charts page.

Reply:
@theashiot WDYT?

Reply:
Just to make it easier for the users, how about rather than mentioning the prerequisite about helm chart, we add the required commands?

### Deploy the Example Application to WildFly on OpenShift

If you haven't already installed the WildFly helm chart, install it:
`helm repo add wildfly https://docs.wildfly.org/wildfly-charts/`

Update the helm chart to ensure you have the latest one:
`helm repo update`

We can deploy our example application to WildFly on OpenShift using the WildFly Helm Chart:

`helm install oidc-app -f /PATH/TO/ELYTRON/EXAMPLES/simple-webapp-saml/charts/helm.yaml wildfly/wildfly
`
...

Reply:
Sounds good, Ashwin and thanks for the suggestion Prarthona!

"Update the helm chart to ensure you have the latest one:" -> I think that could be "If you've already installed the WildFly Helm Chart", be sure to update it to ensure you have the latest one:"

Reply:
thanks, done. Also, moved WildFly helm chart to resources section and added helm installation to prerequisites.